### ISSUE TITLE
Implement `find` for btree_map::Iter

### ISSUE THREAD

**Issue by hellow554** [rust-lang/rust#54550]
A Binarytree should be able to find its key in `O(Log(n))` time, instead is uses the linear implementation from the "generic" Iterator.

There should be an implementation of `find` for [`Iter`](https://doc.rust-lang.org/std/collections/btree_map/struct.Iter.html)

**Comment by Stargateur** [comment]
Isn't the purpose of [`get()`](https://doc.rust-lang.org/std/collections/btree_map/struct.BTreeMap.html#method.get) ?

**Comment by hellow554** [comment]
Not exactly, but you're pointing at the right direction. You can't use the fact that the btree is binary ordered. I don't think that this will work out. :/
